calamus/fields.py

```diff
@@ -342,6 +342,55 @@ class Raw(_JsonLDField, fields.Raw):
     pass
 
 
+class RawJsonLD(_JsonLDField, fields.Raw):
+    """A raw json-ld field."""
+
+    def _dereference_single_id(self, value, attr, **kwargs):
+        """Dereference a single id."""
+        data = kwargs["_all_objects"].get(value, None)
+        if not data:
+            raise ValueError("Couldn't dereference id {id}".format(id=value))
+
+        try:
+            data = dict(data)
+            #data = self._dereference_flattened(data, attr, **kwargs)
+        except (TypeError, ValueError):
+            raise ValueError(f"Couldn't convert value '{data}' to dictionary.")
+
+        if self.reverse:
+            # we need to remove the property from the child when handling reverse nesting
+            del data[attr]
+
+        return data
+
+    def _dereference_flattened(self, value, attr, **kwargs):
+        """Dereference an id or a list of ids."""
+        if isinstance(value, list) or isinstance(value, types.GeneratorType):
+            return [self._dereference_flattened(i, attr, **kwargs) for i in value]
+        if isinstance(value, str):
+            return self._dereference_single_id(value, attr, **kwargs)
+        elif isinstance(value, dict):
+            if len(value.keys()) == 1 and "@id" in value:
+                value = self._dereference_single_id(value["@id"], attr, **kwargs)
+                for k, v in value.items():
+                    if not k.startswith('@'):
```

Comment:
@Panaetius this is for the `@id`, `@type`, `@value` keys, i'm not so sure if there's any other keys with `@` prefix that we might want to dereference

Reply:
There's a lot of fields but I don't think any of them need dereferencing: https://github.com/SwissDataScienceCenter/calamus/blob/master/calamus/utils.py#L28